### PR TITLE
Test OpenTelemetry OTLP Exporter proxy

### PR DESCRIPTION
**Pull request by michalvavrik**
### Summary

test coverage for https://github.com/quarkusio/quarkus/issues/39519

TP: https://github.com/quarkus-qe/quarkus-test-plans/pull/178

Please select the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Dependency update
- [ ] Refactoring
- [ ] Backport
- [x] New scenario (non-breaking change which adds functionality)
- [ ] This change requires a documentation update
- [ ] This change requires execution against OCP (use `run tests` phrase in comment)

### Checklist:
- [x] Methods and classes used in PR scenarios are meaningful
- [x] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

File: monitoring/opentelemetry/src/test/java/io/quarkus/ts/opentelemetry/OpenTelemetryProxyIT.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+                    requestToJaeger
+                            .methodName(reqFromQuarkus.methodName())
+                            .serviceName(reqFromQuarkus.serviceName())
+                            .end(msgFromQuarkus.payload());
+                    // send Jaeger response back to the Quarkus application
+                    requestToJaeger.response()
+                            .onSuccess(h -> h.messageHandler(msg -> reqFromQuarkus.response().endMessage(msg)))
+                            .onFailure(err -> reqFromQuarkus.response().status(GrpcStatus.ABORTED).end());
+                })));
+        return () -> CompletableFuture.allOf(proxyHttpServer.close().toCompletionStage().toCompletableFuture(),
```
Comment (jcarranzan) [inline monitoring/opentelemetry/src/test/java/io/quarkus/ts/opentelemetry/OpenTelemetryProxyIT.java:99]: is not needed add some CountDownLatch to help the concurrency management? I looked something similar here--> https://github.com/eclipse-vertx/vertx-http-proxy/blob/0d7bad50c138fee43baf6f3506d5ebd4ee01ddc5/src/test/java/io/vertx/httpproxy/TestBase.java#L94C11-L94C25

File: monitoring/opentelemetry/src/test/java/io/quarkus/ts/opentelemetry/OpenTelemetryProxyIT.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+                .then()
+                .statusCode(200)
+                .body("data.spans.flatten().findAll { it.operationName == '%s' }.size()".formatted(OPERATION_NAME),
+                        greaterThan(0)));
+    }
+
+    private static Response thenRetrieveTraces() {
+        return RestAssured
+                .given()
+                .queryParam("operation", "GET /hello")
```
Comment (jcarranzan) [inline monitoring/opentelemetry/src/test/java/io/quarkus/ts/opentelemetry/OpenTelemetryProxyIT.java:130]: You have OPERATION_NAME constant, use it pls.

**Review by jcarranzan — COMMENTED** [review]
According your TP, is there also `Proxy enabled / disabled during the runtime` test?

**Comment by michalvavrik** [comment]
> According your TP, is there also Proxy enabled / disabled during the runtime test?

here it is enabled, while disabled is everywhere else; I am happy to test something on disabled proxy if you have an idea, what should I test? I could set proxy options that are going to be ignored if you want

**Comment by jcarranzan** [comment]
> > According your TP, is there also Proxy enabled / disabled during the runtime test?
> 
> here it is enabled, while disabled is everywhere else; I am happy to test something on disabled proxy if you have an idea, what should I test? I could set proxy options that are going to be ignored if you want



> > According your TP, is there also Proxy enabled / disabled during the runtime test?
> 
> here it is enabled, while disabled is everywhere else; I am happy to test something on disabled proxy if you have an idea, what should I test? I could set proxy options that are going to be ignored if you want

It is not needed, thanks. IMHO this is quite enough for our testing coverage purpose.

File: monitoring/opentelemetry/src/test/java/io/quarkus/ts/opentelemetry/OpenTelemetryProxyIT.java (hunk 1 above)
Comment (michalvavrik) [reply monitoring/opentelemetry/src/test/java/io/quarkus/ts/opentelemetry/OpenTelemetryProxyIT.java:99]: if you mean that gRPC client should be closed before HTTP server, I think you would be technically right, not sure it matters, but I'll look into it.

**Comment by michalvavrik** [comment]
> It is not needed, thanks. IMHO this is quite enough for our testing coverage purpose.

Thanks. I will add one little check that proxy enablement flag works.

File: monitoring/opentelemetry/src/test/java/io/quarkus/ts/opentelemetry/OpenTelemetryProxyIT.java (hunk 1 above)
Comment (michalvavrik) [reply monitoring/opentelemetry/src/test/java/io/quarkus/ts/opentelemetry/OpenTelemetryProxyIT.java]: I have chained these 2 operations. I don't have too much of experience with this, hopefully it's done correctly.

**Review by jcarranzan — APPROVED** [review]
LGTM, thanks!.

**Pull request merged**